src/microbit/model/display.py

```diff
@@ -63,19 +62,18 @@ def scroll(self, value, delay=150, wait=True, loop=False, monospace=False):
         while True:
             # Show the scrolled image one square at a time.
             for x in range(appended_image.width() - CONSTANTS.LED_WIDTH + 1):
-                self.__image_lock.acquire()
+                self.__lock.acquire()
                 self.__image.blit(
                     appended_image, x, 0, CONSTANTS.LED_WIDTH, CONSTANTS.LED_HEIGHT
                 )
-                self.__image_lock.release()
 
                 # If show or scroll is called again, there will be a different pid and break
-                self.__thread_identifier_lock.acquire()
                 if self.__current_pid != threading.get_ident():
+                    self.__lock.release()
                     break

```

Comment:
Perhaps consider doing this before updating the display? `__current_pid` might have been changed, but we check after the update.

Reply:
Done 💃